algoliasearch-core/src/main/java/com/algolia/search/models/settings/IndexSettings.java

```diff
@@ -670,6 +672,20 @@ public IndexSettings setCustomSettings(Map<String, Object> customSettings) {
     return this;
   }
 
+  public Boolean getEnablePersonalization() {
+    return enablePersonalization != null && enablePersonalization;
+  }
+
+  public IndexSettings setEnablePersonalization(Boolean enablePersonalization) {
+    this.enablePersonalization = enablePersonalization;
+    return this;
+  }
+
+  @JsonProperty("enablePersonalization")
+  private Boolean internalEnablePersonalization() {
```

Comment:
I wonder if this is a good approach? It's just a way to not serialize/deserialize then the property when not set, however, still get the default value `false` from the getter when the property is not set.

What do you think? we can also always serialize the property, but not sure if it is the wanted behavior!

Reply:
I don't think we need that because `enablePersonalization` is a `Boolean`, an Object and thus nullabe.
At the top of the class we have set `JSON.Include_non_null` so all null values won't be serialized. A simple getter/setter will work for that case, like `advancedSyntax` for example:

https://github.com/algolia/algoliasearch-client-java-2/blob/147aea5f277db9016afe48516fae8fb8ec5ef493/algoliasearch-core/src/main/java/com/algolia/search/models/settings/IndexSettings.java#L357-L364

Does that answer your question?

Reply:
> The needed change is to expose the enablePersonalization boolean as a
valid setting parameter, which defaults to **false** when missing.

It is okay then to not have a default value ?

Reply:
Yes! The field won't be serialized and and then engine won't set it internally.

Reply:
Nice! even better then :D
Thank you!